core/trino-main/src/main/java/io/trino/sql/planner/optimizations/BeginTableWrite.java

```diff
@@ -184,7 +184,7 @@ public PlanNode visitMergeWriter(MergeWriterNode mergeNode, RewriteContext<Optio
             MergeTarget mergeTarget = (MergeTarget) getContextTarget(context);
             return new MergeWriterNode(
                     mergeNode.getId(),
-                    rewriteModifyTableScan(mergeNode.getSource(), mergeTarget.getHandle()),
```

Comment:
if so, the Merge's TS should not return true from `io.trino.sql.planner.plan.TableScanNode#isUpdateTarget`

Reply:
As you may have gathered from the comment below, I was wrong about needing to find the right `TableScanNode` **if it exists**.